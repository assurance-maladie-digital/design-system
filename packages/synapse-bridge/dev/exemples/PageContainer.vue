<template>
	<PageContainer>
		<VCard class="pa-8">
			Contenu de la page.
		</VCard>
	</PageContainer>
</template>

<script lang="ts" setup>
	import PageContainer from '../../src/elements/PageContainer'
</script>
