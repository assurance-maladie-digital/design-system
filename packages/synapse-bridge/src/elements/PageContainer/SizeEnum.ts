const SizeEnum = {
	X_LARGE:'xl',
	LARGE:'l',
	MEDIUM: 'm',
	SMALL: 's'
}

 const SIZE_ENUM_VALUES = Object.values(SizeEnum)

export { SizeEnum, SIZE_ENUM_VALUES }
