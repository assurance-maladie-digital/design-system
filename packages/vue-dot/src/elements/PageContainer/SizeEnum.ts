export enum SizeEnum {
	X_LARGE = 'xl',
	LARGE = 'l',
	MEDIUM = 'm',
	SMALL = 's'
}

export const SIZE_ENUM_VALUES = Object.values(SizeEnum);
