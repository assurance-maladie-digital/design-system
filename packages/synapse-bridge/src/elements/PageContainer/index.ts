import PageContainer from './PageContainer.vue';

export default PageContainer;
