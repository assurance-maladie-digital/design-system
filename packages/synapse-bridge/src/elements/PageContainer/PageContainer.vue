<script lang="ts">
	import { defineComponent } from 'vue'

	import { propValidator } from '@/helpers/propValidator'

	import { SizeEnum, SIZE_ENUM_VALUES } from './SizeEnum'

	export default defineComponent({
		props: {
			size: {
				type: String,
				default: SizeEnum.X_LARGE,
				validator: (value: string) => propValidator('size', SIZE_ENUM_VALUES, value)
			},
			spacing: {
				type: String,
				default: undefined
			},
			color: {
				type: String,
				default: 'transparent'
			}
		},
		data() {
			return {
				props: {
					size: this.size,
					spacing: this.spacing,
					color: this.color
				},
			}
		},
		computed: {
			spacingClass(): string {
				if (this.spacing) {
					return this.spacing
				}

				const spacingMapping: Record<string, string> = {
					xs: 'px-0',
					sm: 'px-4',
					md: 'px-8',
					lg: 'px-8',
					xl: 'px-8'
				}

				const spacing = spacingMapping[this.$vuetify.display.name]

				return `py-10 ${spacing}`
			},

			containerSize(): number {
				const sizeMapping: Record<string, number> = {
					[SizeEnum.X_LARGE]: 1440,
					[SizeEnum.LARGE]: 960,
					[SizeEnum.MEDIUM]: 800,
					[SizeEnum.SMALL]: 600
				}

				return sizeMapping[this.size]
			}
		}
	})
</script>

<template>
	<div :class="spacingClass" class="vd-page-container d-flex justify-center">
		<VSheet :width="containerSize" :color="color">
			<slot />
		</VSheet>
	</div>
</template>

<style lang="scss" scoped>
.vd-page-container {
	flex: 1;
}
</style>
