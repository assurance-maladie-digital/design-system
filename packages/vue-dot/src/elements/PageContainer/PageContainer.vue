<template>
	<div
		:class="spacingClass"
		class="vd-page-container d-flex justify-center"
	>
		<VSheet
			:width="containerSize"
			:color="color"
		>
			<slot />
		</VSheet>
	</div>
</template>

<script lang="ts">
	import Vue from 'vue';
	import Component, { mixins } from 'vue-class-component';

	import { SizeEnum, SIZE_ENUM_VALUES } from './SizeEnum';

	import { IndexedObject } from '../../types';

	import { propValidator } from '../../helpers/propValidator';

	const Props = Vue.extend({
		props: {
			size: {
				type: String,
				default: SizeEnum.X_LARGE,
				validator: (value: string) => propValidator('size', SIZE_ENUM_VALUES, value)
			},
			spacing: {
				type: String,
				default: undefined
			},
			color: {
				type: String,
				default: 'transparent'
			}
		}
	});

	const MixinsDeclaration = mixins(Props);

	@Component
	export default class PageContainer extends MixinsDeclaration {
		get spacingClass(): string {
			if (this.spacing) {
				return this.spacing;
			}

			const spacingMapping: IndexedObject<string> = {
				xs: 'px-0',
				sm: 'px-4',
				md: 'px-8',
				lg: 'px-8',
				xl: 'px-8'
			};

			const spacing = spacingMapping[this.$vuetify.breakpoint.name];

			return `py-10 ${spacing}`;
		}

		get containerSize(): number {
			const sizeMapping: IndexedObject<number> = {
				[SizeEnum.X_LARGE]: 1440,
				[SizeEnum.LARGE]: 960,
				[SizeEnum.MEDIUM]: 800,
				[SizeEnum.SMALL]: 600
			};

			return sizeMapping[this.size];
		}
	}
</script>
