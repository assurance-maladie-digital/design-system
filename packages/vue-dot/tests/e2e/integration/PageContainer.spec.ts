import PageContainer from '@/elements/PageContainer';

// Tests
describe('PageContainer', () => {
	it('renders correctly', () => {
		// Mount component
		cy.mountComponent(PageContainer);
	});
});
