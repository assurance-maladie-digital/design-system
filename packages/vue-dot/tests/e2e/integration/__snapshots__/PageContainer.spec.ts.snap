exports[`PageContainer > renders correctly #0`] = `
<div id="__cy_root" class="v-application" data-app="true">
  <div class="vd-page-container d-flex justify-center py-10 px-0">
    <div class="v-sheet theme--light transparent" style="width: 1440px;"></div>
  </div>
</div>
`;
